src/rust/runtime/condition_variable.rs

```diff
@@ -131,17 +131,22 @@ impl Future for YieldPoint {
     /// The first time that this future is polled, it is not ready but the next time must be a signal so then it is
     /// ready.
     fn poll(self: Pin<&mut Self>, context: &mut Context) -> Poll<Self::Output> {
+        let state: YieldState = self.state;
+        let num_ready: usize = self.cond_var.num_ready;
         let self_: &mut Self = self.get_mut();
-        if self_.cond_var.num_ready > 0 {
-            self_.cond_var.num_ready -= 1;
-            Poll::Ready(())
-        } else {
-            if self_.state == YieldState::Running {
+        match state {
+            YieldState::Running => {
                 self_.cond_var.add_waiter(self_.id, context.waker().clone());
-            }
-            self_.state = YieldState::Yielded;
-            Poll::Pending
-        }
+                self_.state = YieldState::Yielded;
+            },
+            YieldState::Yielded if num_ready > 0 => {
+                self_.cond_var.num_ready -= 1;
+                self_.state = YieldState::Running;
+                return Poll::Ready(());
+            },
+            _ => (),
```

Comment:
This line can be changed to `_ => Poll::Pending` to make all return statements coherent within the match.

Reply:
sure, makes sense